deepchem/splits/splitters.py

```diff
@@ -1588,10 +1594,11 @@ def generate_scaffolds(self,
             if ind % log_every_n == 0:
                 logger.info("Generating scaffold %d/%d" % (ind, data_len))
             scaffold = _generate_scaffold(smiles)
-            if scaffold not in scaffolds:
-                scaffolds[scaffold] = [ind]
-            else:
-                scaffolds[scaffold].append(ind)
+            if scaffold is not None:
```

Comment:
What happened previously in these cases?

Reply:
Previously, the code will not execute (ref: #3512) - but now, it can execute.